core/src/com/unciv/logic/GameInfo.kt

```diff
@@ -265,11 +267,62 @@ class GameInfo {
             // This doesn't HAVE to go here, but why not.
             // As of version 3.1.3, trade offers of "Declare war on X" and "Introduction to X" were changed to X,
             //   with the extra text being added only on UI display (solved a couple of problems).
-            for(trade in civInfo.tradeRequests.map { it.trade })
-                for(offer in trade.theirOffers.union(trade.ourOffers)){
+            for(trade in civInfo.tradeRequests.map { it.trade }) {
+                for (offer in trade.theirOffers.union(trade.ourOffers)) {
                     offer.name = offer.name.removePrefix("Declare war on ")
                     offer.name = offer.name.removePrefix("Introduction to ")
                 }
+            }
+
+            // As of 3.4.9 cities are referenced by id, not by name
+            // So try to update every tradeRequest (if there are no conflicting names)
+            for(tradeRequest in civInfo.tradeRequests) {
+                val trade = tradeRequest.trade
+                val toRemove = ArrayList<TradeOffer>()
+                for (offer in trade.ourOffers) {
```

Comment:
Better to do this as for (offer in trade.ourOffers.toList()) which creates a new list, and then you can do trade.ourOffers.remove() directly and can forgo the toRemove list